io-engine/src/bdev/nexus/nexus_channel.rs

```diff
@@ -220,6 +222,18 @@ impl<'n> NexusChannel<'n> {
                     writers.push(w);
                     readers.push(r);
                 }
+                (
+                    Err(CoreError::OpenBdev {
+                        source: _,
+                    }),
+                    _,
+                )
+                | (
+                    _,
+                    Err(CoreError::OpenBdev {
+                        source: _,
+                    }),
+                ) => ret = Errno::EAGAIN as i32,
```

Comment:
Could be be re-written in a more functional style collecting Results into vec, rather than having reads,writers and ret?